test/RibbonV2SharePrice.test.ts

```diff
@@ -0,0 +1,134 @@
+import { artifacts, ethers, waffle } from 'hardhat';
+import { expect } from 'chai';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address';
+import { MockCozyToken, MockRibbonVaultV2, RibbonV2SharePrice } from '../typechain';
+
+const { deployContract } = waffle;
+const { parseUnits } = ethers.utils;
+
+describe('RibbonV2SharePrice', function () {
+  let deployer: SignerWithAddress, recipient: SignerWithAddress;
+  let mockRUsdc: MockRibbonVaultV2;
+  let trigger: RibbonV2SharePrice;
+  let triggerParams: (string | number[])[] = []; // trigger params deployment parameters
+
+  before(async () => {
+    [deployer, recipient] = await ethers.getSigners();
+  });
+
+  beforeEach(async () => {
+    // Deploy Mock Ribbon Vault
+    const mockRibbonVaultY2Artifact = await artifacts.readArtifact('MockRibbonVaultV2');
+    mockRUsdc = <MockRibbonVaultV2>await deployContract(deployer, mockRibbonVaultY2Artifact);
```

Comment:
Related to the above comment, we should delete the Mock contract and instead have tests modify storage of forked mainnet contracts, as we do in `YearnCrvTwoTokens.test.ts`. Modifying mainnet storage instead of using a mock will give more confidence in the contract/tests, since it replaces the mock with real mainnet contracts/state.

More generally, it seems this test file was based on an older test file from before Hardhat allowed you to arbitrarily modify storage, whereas `YearnCrvTwoTokens.test.ts` is a newer test file that leverages that feature